src/lib.rs

```diff
@@ -74,13 +70,15 @@ impl XenControl {
         last_error!(xc, hvm_cpu)
     }
 
-    pub fn monitor_enable(&mut self, domid: u32) -> Result<&PageInfo> {
+    pub fn monitor_enable(&mut self, domid: u32) -> Result<(*mut c_void, u32)> {
         let xc = self.handle.as_ptr();
-        let ring_page = unsafe {
+        let domid_compat: u16 = domid.try_into().unwrap();
+        let mut remote_port: u32 = 0;
+        let ring_page: *mut c_void = unsafe {
             xc_clear_last_error(xc);
-            xc_monitor_enable(xc, domid.try_into().unwrap(), &mut self.evtchn_port as _) as *const PageInfo
```

Comment:
Why did you do this? This change reintroduces issue https://github.com/Wenzel/libmicrovmi/issues/20 :(